base/set.jl

```diff
@@ -1,5 +1,19 @@
 # This file is a part of Julia. License is MIT: https://julialang.org/license
 
+# AbstractSet
+
+# AbstractSets are idempotent under indexing:
+keys(set::AbstractSet) = set
+@inline function getindex(set::AbstractSet, x)
```

Comment:
Why is this change in this PR?  Can't we have a PR that just makes `KeyIterator` into `KeySet` without adding `getindex`?

Reply:
Yes, agreed, I decided to decouple this but haven't got to it just yet